halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/services/v1/PluginService.java

```diff
@@ -133,6 +136,17 @@ public void setPluginsDownloadingEnabled(
         deploymentService.getDeploymentConfiguration(deploymentName);
     Plugins plugins = deploymentConfiguration.getPlugins();
     plugins.setDownloadingEnabled(enable);
+    Map<String, List<Map>> initContainers = new HashMap<>();
+    List<Map> containers = new ArrayList();
+    Map pluginDownloadingContainer = new HashMap();
+    pluginDownloadingContainer.put("name", "pluginDownloader");
+    pluginDownloadingContainer.put("image", "pluginDownloader");
+    List<String> command = new ArrayList<>();
+    command.add("ls");
+    pluginDownloadingContainer.put("command", command);
+    containers.add(pluginDownloadingContainer);
+    initContainers.put("spin-deck", containers);
+    deploymentConfiguration.getDeploymentEnvironment().setInitContainers(initContainers);
```

Comment:
creates an init container successfully, in deck

Reply:
Is it necessary to add commands to the init container? If you can avoid it, then there's no need to change halyard code, as it already supports adding init containers to services just by specifying the image (and volumes) to use in Hal configs.
If you absolutely need to add commands, it would be better to extend existing init containers logic to customize commands, by specifying them in Hal configs.

Reply:
its not, but the initContainer should be addded /removed depending on whether pluginDownloading is enabled/disabled